src/main/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/oidc/OIDCLoginBackingBean.java

```diff
@@ -0,0 +1,169 @@
+package edu.harvard.iq.dataverse.authorization.providers.oauth2.oidc;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import org.omnifaces.util.Faces;
+
+import edu.harvard.iq.dataverse.DataverseSession;
+import edu.harvard.iq.dataverse.UserServiceBean;
+import edu.harvard.iq.dataverse.authorization.AuthenticatedUserDisplayInfo;
+import edu.harvard.iq.dataverse.authorization.AuthenticationServiceBean;
+import edu.harvard.iq.dataverse.authorization.users.AuthenticatedUser;
+import edu.harvard.iq.dataverse.authorization.providers.oauth2.OAuth2FirstLoginPage;
+import edu.harvard.iq.dataverse.authorization.providers.oauth2.OAuth2UserRecord;
+import edu.harvard.iq.dataverse.authorization.UserRecordIdentifier;
+import edu.harvard.iq.dataverse.util.SystemConfig;
+import fish.payara.security.openid.api.JwtClaims;
+import fish.payara.security.openid.api.OpenIdConstant;
+import fish.payara.security.openid.api.OpenIdContext;
+import jakarta.ejb.EJB;
+import jakarta.ejb.Stateless;
+import jakarta.inject.Inject;
+import jakarta.inject.Named;
+import jakarta.json.JsonString;
+import jakarta.json.JsonValue;
+
+/**
+ * Backing bean of the OIDC login process. Used from the login and the
+ * callback pages.
+ */
+@Stateless
+@Named
+public class OIDCLoginBackingBean implements Serializable {
+    private static final Logger logger = Logger.getLogger(OIDCLoginBackingBean.class.getName());
+
+    @EJB
+    AuthenticationServiceBean authenticationSvc;
+
+    @EJB
+    SystemConfig systemConfig;
+
+    @EJB
+    UserServiceBean userService;
+
+    @Inject
+    DataverseSession session;
+
+    @Inject
+    OAuth2FirstLoginPage newAccountPage;
+
+    @Inject
+    OpenIdContext openIdContext;
+
+    /**
+     * Generate the OIDC log in link.
+     */
+    public String getLogInLink(final OIDCAuthProvider oidcAuthProvider) {
+        final UserRecordIdentifier userRecordIdentifier = getUserRecordIdentifier();
+        if (userRecordIdentifier != null) {
+            setUser();
+            return SystemConfig.getDataverseSiteUrlStatic();
+        }
+        return "/oidc/login?target=JSF&oidcp=" + oidcAuthProvider.getId();
+    }
+
+    /**
+     * View action for callback.xhtml, the browser redirect target for the OAuth2
+     * provider.
+     * 
+     * @throws IOException
+     */
+    public void setUser() {
+        try {
+            final String subject = openIdContext.getSubject();
+            final OIDCAuthProvider provider = getProvider();
+            final UserRecordIdentifier userRecordIdentifier = new UserRecordIdentifier(provider.getId(), subject);
+            AuthenticatedUser dvUser = authenticationSvc.lookupUser(userRecordIdentifier);
+            if (dvUser == null) {
+                if (!systemConfig.isSignupDisabledForRemoteAuthProvider(provider.getId())) {
```

Comment:
There is no else here - what happens if this is a new user and the signup is disabled? Where do people end up? Is there a nice 401 error page explaining what is going on?

Reply:
I do not know... It is just copy-paste from the original code...